katcp/core.py

```diff
@@ -299,7 +299,23 @@ def format_argument(self, arg):
         elif isinstance(arg, bool):
             return str(int(arg))
         else:
-            return str(arg)
+            try:
+                return str(arg)
+            except Exception:
+                # unicode characters will break the str cast, so
+                # try to encode to ascii and replace the offending characters
+                # with a '?' character
+                logger.exception("Error casting message argument to str!")
+                try:
+                    if type(arg) != unicode:
```

Comment:
`isinstance` is generally preferable to exact type comparisons.

Reply:
Fair, I knew this at some stage. Changing to use `isinstance`.